File: Guvi_prime_fermat_13.py
def prime(num):
        pw=2**num
        fer=pw-2
        if(fer%num==0):
            print("Prime no. ")
        else:
            print("Not a Prime no. ")


def main():
    n=int(input("Enter the number to be checked"))
    prime(n)
    
if __name__=="__main__":
    main()

File: Guvi_prime_fermat_13.java
import java.io.*;
import java.util.*;

class Guvi_prime_fermat_13
{
	public static void main(String args[])
	{
		
		Scanner sc=new Scanner(System.in);
		
		int num;
		
		long pw,fer;
		
		num=sc.nextInt();
		
		pw=(long)(Math.pow(2,num));
		
		fer=pw-2;
		
		if(fer%num==0)
			System.out.println("No. is Prime");
		else
			System.out.println("No. is not Prime");
	}
}